### ISSUE TITLE
Tkinter causes kernel to restart if backend is not set to "Inline"

### ISSUE THREAD

**Issue by battaglia01** [spyder-ide/spyder#17124]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [x] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [x] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
To reproduce this:

1. Start with a clean Spyder install
2. Go to Preferences -> IPython Console -> Graphics and set "Backend" to anything except "Inline." Let's say Qt5.
3. At this point, close the entire program and restart it, so that it starts up with Qt5 as the backend when loading the first kernel.
4. Type `import tkinter; tkinter.Tk()`

You then get the following message
```
____________________________________________________________
Restarting kernel...
 ```

and the kernel restarts.

This doesn't seem to happen if you have "Inline" set, for whatever reason, and it also doesn't happen if you have "Inline" set and then change it without restarting the kernel, so it's recommended to try the above method to reproduce it.

I searched for issues involving Tkinter but this seems different. For instance, in #17061 we have an error which causes a traceback. There are many similar issues with the same traceback. But, in this situation, there is no traceback, just this "Restarting kernel..." message. I tried the solution that spyder-bot suggested in #17061, which was to `conda install jupyter_client=6.1.12 freetype=2.10.4`, but it made no difference.

## Versions

Python 3.9.7 64-bit | Qt 5.9.7 | PyQt5 5.9.2 | Darwin 17.7.0 
on MacOS High Sierra 10.13.6

### Dependencies

```

# Mandatory:
applaunchservices >=0.1.7     :  0.2.1 (OK)
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.2 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.29.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.0 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.4.0 (OK)
nbconvert >=4.0               :  6.1.0 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
parso >=0.7.0;<0.9.0          :  0.8.2 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.10.0 (OK)
pylint >=2.5.0;<2.10.0        :  2.9.6 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.2.2;<1.3.0          :  1.2.4 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=1.0.2             :  1.0.3 (OK)
qtconsole >=5.1.0             :  5.1.1 (OK)
qtpy >=1.5.0                  :  1.10.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  58.0.4 (OK)
sphinx >=0.6.6                :  4.2.0 (OK)
spyder_kernels >=2.1.1;<2.2.0 :  2.1.3 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
zmq >=17                      :  22.3.0 (OK)

# Optional:
cython >=0.21                 :  0.29.25 (OK)
matplotlib >=2.0.0            :  3.4.3 (OK)
numpy >=1.7                   :  1.21.2 (OK)
pandas >=1.1.1                :  1.3.4 (OK)
scipy >=0.17.0                :  1.7.1 (OK)
sympy >=0.7.3                 :  1.9 (OK)
```

**Comment by ccordoba12** [comment]
Hey @battaglia01, thanks for reporting. I don't think you can have two graphical event loops running at the same time in a single console, i.e. Qt5 (for the Matplotlib Qt5 backend) and Tk (with the code you evaluated).

That's why Matplotlib prevents setting a new interactive backend (e.g. Tk) after you set a previous one of the same kind (Qt5):

![imagen](https://user-images.githubusercontent.com/365293/148101128-ce4359af-1dbc-4940-9a7e-5dc99ff84cac.png)

>     Type import tkinter; tkinter.Tk()
> and the kernel restarts.

It seems that trying to do that with code (as you are) causes the kernel to crash.

> This doesn't seem to happen if you have "Inline" set

The exception is Inline because because it doesn't require a graphical toolkit to run; it just generates static images.

**Comment by ccordoba12** [comment]
So the solution is to use Inline or the same Matplotlib backend you want to develop apps for. And I don't think there's a workaround for that.

Let me know if I can help you with something else.

**Comment by ccordoba12** [comment]
Closing due to lack of response.

**Comment by battaglia01** [comment]
@ccordoba12 sorry for the slow response. This doesn't work even if I set it to Tkinter, restart the program and load it with Tkinter already loaded. Same thing:

```
import tkinter; tkinter.Tk()

Restarting kernel...
```

**Comment by ccordoba12** [comment]
Problems with the Tkinter backend should be fixed in 5.2.2, to be released at the end of the week.
